### ISSUE TITLE
Wrong read speed result for Samsung SSD 970 EVO Plus 

### ISSUE THREAD

**Issue by Vascom** [JonMagon/KDiskMark#48]
- Linux-distro: Fedora 33
- Desktop Environment (KDE/GNOME etc.): KDE
- Qt Version: 5.15.2
- KDiskMark Version: 2.2.0
- FIO Version: fio-3.21

### Description:
Strange speed result for Samsung SSD 970 EVO Plus (1TB and others).
By datasheet and in gnome-disk-utility I have ~3500MB/s sequential read but kdiskmark give me only ~1500MB/s.
And write speed ~2900MB/s. So read speed seems not relevant.
 
### Steps To Reproduce:
Run test.

Can you help solve this bug?

**Comment by tim77** [comment]
Same issue. When i've tested first time 5 months ago KDiskMark shows real speed - 3500MB/s:

![Снимок экрана от 2020-10-24 09-18-32](https://user-images.githubusercontent.com/5614476/110232949-ab582780-7f29-11eb-9c36-3e4061d15b7e.png)

Now:
![Снимок экрана от 2021-03-06 12-10-00](https://user-images.githubusercontent.com/5614476/110232966-c9be2300-7f29-11eb-8705-a4dcba0ab663.png)

**Comment by JonMagon** [comment]
Try to disable the flush cache option in Settings. This may not work as expected. If it helps, I will disable this option by default.

**Comment by Vascom** [comment]
Disabling flush cache option not helped.

**Comment by JonMagon** [comment]
Could you find the release the issue appeared the first time?

**Comment by tim77** [comment]
As far i know @Vascom already tried this. I thought as well that this issue appears in some KDiskMark updates but seems like something else causing it.

**Comment by JonMagon** [comment]
I can't reproduce it, because I haven't such an SSD, but it's definitely a bug.

**Comment by tim77** [comment]
Similar discussion on [Reddit](https://www.reddit.com/r/linuxhardware/comments/leprjs/samsung_970_evo_plus_250gb_giving_less_than/) and the same disk. Doubt that this could help but a least seems like popular issue.

**Comment by Vascom** [comment]
May be we can give you some debug output.

**Comment by JonMagon** [comment]
Can you please check with [2.0.0](https://github.com/JonMagon/KDiskMark/releases/tag/2.0.0)?

**Comment by tim77** [comment]
Tested with `2.0.0` and `1.6.2` and :100: the same issue. Maybe this the kernel bug or something? Very weird since my first test on the same OS and same SSD was OK.

**Comment by JonMagon** [comment]
Then it may be caused by separation loops inside kdiskmark. Could you please try also [1.6.0](https://github.com/JonMagon/KDiskMark/releases/tag/1.6.0)?

**Comment by Vascom** [comment]
The same with 1.6.0.
![Screenshot_20210308_083000](https://user-images.githubusercontent.com/550017/110278580-8c6c9a80-7fe8-11eb-9d87-803b9605fdd5.png)

**Comment by JonMagon** [comment]
Then it can be really not kdiskmark issue.

**Comment by JonMagon** [comment]
What is the result of the command below?
```bash
fio --ioengine=libaio --direct=1 --randrepeat=0 --refill_buffers --end_fsync=1 --filename=test-fio.tmp --name=readjob --size=128M --bs=1m --rw=read --iodepth=8 --loops=5
```

**Comment by Vascom** [comment]
```
Last login: Tue Mar  9 12:26:53 2021 from 10.9.0.1
[vascom@milkyway ~]$ fio --ioengine=libaio --direct=1 --randrepeat=0 --refill_buffers --end_fsync=1 --filename=test-fio.tmp --name=readjob --size=128M --bs=1m --rw=read --iodepth=8 --loops=5
readjob: (g=0): rw=read, bs=(R) 1024KiB-1024KiB, (W) 1024KiB-1024KiB, (T) 1024KiB-1024KiB, ioengine=libaio, iodepth=8
fio-3.21
Starting 1 process
readjob: Laying out IO file (1 file / 128MiB)

readjob: (groupid=0, jobs=1): err= 0: pid=4886: Tue Mar  9 12:45:11 2021
  read: IOPS=1376, BW=1376MiB/s (1443MB/s)(640MiB/465msec)
    slat (usec): min=398, max=2304, avg=715.00, stdev=285.00
    clat (usec): min=313, max=14322, avg=4882.13, stdev=1889.90
     lat (usec): min=890, max=16629, avg=5598.28, stdev=2146.95
    clat percentiles (usec):
     |  1.00th=[ 1029],  5.00th=[ 4015], 10.00th=[ 4113], 20.00th=[ 4228],
     | 30.00th=[ 4293], 40.00th=[ 4359], 50.00th=[ 4490], 60.00th=[ 4621],
     | 70.00th=[ 4752], 80.00th=[ 4948], 90.00th=[ 5080], 95.00th=[11076],
     | 99.00th=[11731], 99.50th=[13042], 99.90th=[14353], 99.95th=[14353],
     | 99.99th=[14353]
  lat (usec)   : 500=0.78%, 1000=0.16%
  lat (msec)   : 2=1.41%, 4=2.50%, 10=88.59%, 20=6.56%
  cpu          : usr=1.29%, sys=81.72%, ctx=6, majf=0, minf=2059
  IO depths    : 1=0.8%, 2=1.6%, 4=3.1%, 8=94.5%, 16=0.0%, 32=0.0%, >=64=0.0%     submit    : 0=0.0%, 4=100.0%, 8=0.0%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
     complete  : 0=0.0%, 4=99.2%, 8=0.8%, 16=0.0%, 32=0.0%, 64=0.0%, >=64=0.0%
     issued rwts: total=640,0,0,0 short=0,0,0,0 dropped=0,0,0,0
     latency   : target=0, window=0, percentile=100.00%, depth=8

Run status group 0 (all jobs):
   READ: bw=1376MiB/s (1443MB/s), 1376MiB/s-1376MiB/s (1443MB/s-1443MB/s), io=640MiB (671MB), run=465-465msec
[vascom@milkyway ~]$
```

**Comment by JonMagon** [comment]
I would guess that this is the real reading performance.
What if test the speed with dd? Copy the all output.
```bash
dd if=/dev/zero of=test-fio.tmp bs=1M count=128
echo 3 | sudo tee /proc/sys/vm/drop_caches
dd if=test-fio.tmp of=/dev/null bs=1M
```

**Comment by tim77** [comment]
```
echo 3 | sudo tee /proc/sys/vm/drop_caches
dd if=test-fio.tmp of=/dev/null bs=1M
128+0 записей получено
128+0 записей отправлено
134217728 байт (134 MB, 128 MiB) скопирован, 0,0916864 s, 1,5 GB/s
3
128+0 записей получено
128+0 записей отправлено
134217728 байт (134 MB, 128 MiB) скопирован, 0,0393467 s, 3,4 GB/s
```

**Comment by JonMagon** [comment]
No idea, something wrong with fio.

**Comment by JonMagon** [comment]
By the way, @tim77 haven't you changed partitions since the results was correct? Maybe alignment is wrong...

**Comment by tim77** [comment]
> By the way, @tim77 haven't you changed partitions since the results was correct? Maybe alignment is wrong...

Indeed. Partition was changed since first time. BTW i've also wondered about some another issue in BTRFS and i assumed is alignment can be the cause of it or not https://pagure.io/fedora-btrfs/project/issue/36#comment-701576

**Comment by Vascom** [comment]
I just found that it is seems not kdiskmark problem.
At 5.9.16 kernel I see full speed. But on 5.10-5.11.5 still half speed.

**Comment by JonMagon** [comment]
I think you should open an issue in the [fio repository](https://github.com/axboe/fio), I can't resolve it.

**Comment by tim77** [comment]
In same time `gnome-disks` shows close to speed which rated by datasheet, but also clearly shown that it's half at beginning:  
![Снимок экрана от 2021-03-09 22-10-33](https://user-images.githubusercontent.com/5614476/110633704-c2a94600-81b1-11eb-8043-04b8cb1467a0.png)

But i am definitely not trying to say that this is `KDiskMark` bug, no. But for end user this could not obvious.

**Comment by tim77** [comment]
Yep. We will try to file a bug into kernel and fio. Worth to try. At least maybe we can learn something new and maybe this is _a feature_.
